### PR TITLE
MorphoTagger and Vocabaluries refactoring

### PR DESCRIPTION
**Pull request by acriptis**
- `MorphoTagger` refactored for usage of the `SimpleVocabulary` (old MorphoTagger models need no retrain and fully compatible with `SimpleVocabulary`).
- `DefaultVocabulary` is deprecated and deleted
- `LowercasePreprocessor` renamed into `CharSplittingLowercasePreprocessor`. The latter documented with example.
- Bug in configs of `seq2seq_go_bot` fixed

### CODE REVIEW

File: deeppavlov/models/preprocessors/capitalization.py [hunk 1]
```diff
@@ -84,7 +86,15 @@ def process_word(word: str, to_lower: bool = False,
...
-        a preprocessed word
+        a preprocessed word.
+
+    Example:
+        Input params example:
+            word="Zaman",
+            to_lower=True,
+            append_case="first"
+        Output example:
+            ('<FIRST_UPPER>', 'z', 'a', 'm', 'a', 'n')
```
Comment (yoptar) [inline deeppavlov/models/preprocessors/capitalization.py]: A good example of usage examples: https://github.com/deepmipt/DeepPavlov/blob/master/deeppavlov/models/embedders/bow_embedder.py

File: deeppavlov/core/data/simple_vocab.py [hunk 2]
```diff
@@ -29,7 +29,16 @@
...
-    """Implements simple vocabulary."""
+    """Implements simple vocabulary.
+
+    Parameters:
+        special_tokens: tuple of tokens that shouldn't be counted.
+        max_tokens: TODO
+        min_freq: minimal count of a token (except special tokens).
+        pad_with_zeros: TODO
+        unk_token: label assigned to unknown tokens.
+        freq_drop_load: TODO
```
Comment (yoptar) [inline deeppavlov/core/data/simple_vocab.py]: Who can update this?

File: deeppavlov/models/preprocessors/capitalization.py [hunk 3]
```diff
@@ -108,8 +118,8 @@ def process_word(word: str, to_lower: bool = False,
     return tuple(answer)
 
 
-@register('lowercase_preprocessor')
-class LowercasePreprocessor(Component):
+@register('char_splitting_lowercase_preprocessor')
+class CharSplittingLowercasePreprocessor(Component):
```
Comment (yoptar) [inline deeppavlov/models/preprocessors/capitalization.py]: The name is really long. Can we come up with something better?

File: deeppavlov/models/preprocessors/capitalization.py (hunk 1 above)
Comment (acriptis) [reply deeppavlov/models/preprocessors/capitalization.py]: Thanks, updated code according to template

File: deeppavlov/models/preprocessors/capitalization.py (hunk 3 above)
Comment (acriptis) [reply deeppavlov/models/preprocessors/capitalization.py]: How many words is ok? 
PS. It is a holy war about short names vs. precise names

Comment (acriptis) [reply deeppavlov/models/preprocessors/capitalization.py]: Here is a problem that we have a fusion of operations 
- splitting into characters and 
- lowercasing data with decoration of sequence by capitalization prefix (or optionally suffix).

File: deeppavlov/core/data/simple_vocab.py (hunk 2 above)
Comment (acriptis) [reply deeppavlov/core/data/simple_vocab.py]: May be @kudep ? He may know the best description of the params as he authored the `__init__`'s signature

Comment (acriptis) [reply deeppavlov/core/data/simple_vocab.py]: Documented, thanks to @kudep

Comment (kudep) [reply deeppavlov/core/data/simple_vocab.py]: I transferred all my mind's about that to @acriptis, now he can description it))

**Comment by yoptar** [comment]
@AlexeySorokin, could you update docstrings for `CharSplittingLowercasePreprocessor` and `process_word` in [capitalization.py](https://github.com/deepmipt/DeepPavlov/blob/vocabs_refactoring/deeppavlov/models/preprocessors/capitalization.py) to include allowed options for the `append_case` argument?

File: deeppavlov/models/preprocessors/capitalization.py (hunk 3 above)
Comment (AlexeySorokin) [reply deeppavlov/models/preprocessors/capitalization.py]: Check you changes once more for typos and grammar: THE foLLowing operations

**Pull request merged**